File: Android/src/main/java/com/conradkramer/wallet/android/AssetBalanceView.kt
@file:OptIn(ExperimentalMaterial3Api::class)

package com.conradkramer.wallet.android

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.ElevatedCard
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.conradkramer.wallet.viewmodel.Asset

@Composable
fun AssetBalanceView(asset: Asset) {
    ElevatedCard {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(16.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(horizontalAlignment = Alignment.Start) {
                Text(
                    text = asset.balance.currency.name,
                    textAlign = TextAlign.Start,
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 18.sp
                )
                Text(
                    text = asset.balanceString,
                    textAlign = TextAlign.End
                )
            }
            Text(
                text = asset.fiatBalanceString,
                textAlign = TextAlign.End
            )
        }
    }
}

File: Apple/UICore/AssetBalanceView.swift
import Shared
import SwiftUI

public struct AssetBalanceView: View {
    public var asset: Shared.Asset

    public init(asset: Shared.Asset) {
        self.asset = asset
    }

    public var body: some View {
        HStack {
            VStack(alignment: .leading) {
                Text(asset.balance.currency.name)
                    .font(.title2)
                Text(asset.balanceString)
                    .font(.body)
                    .foregroundColor(.gray)
            }
            Spacer()
            Text(asset.fiatBalanceString)
                .font(.body)
        }
        .padding()
        .frame(maxHeight: 64)
    }
}
